src/java.desktop/macosx/classes/com/apple/laf/AquaButtonUI.java

```diff
@@ -326,6 +326,20 @@ public void paint(final Graphics g, final JComponent c) {
         }
     }
 
+    protected void paintFocus(Graphics g, AbstractButton b,
+                              Rectangle viewRect, Rectangle textRect, Rectangle iconRect) {
+        if (g instanceof Graphics2D) {
+            ((Graphics2D)g).setStroke(new BasicStroke(3));
+            ((Graphics2D)g).setRenderingHint(
+                    RenderingHints.KEY_ANTIALIASING,
+                    RenderingHints.VALUE_ANTIALIAS_ON);
+
+        }
+        Color ringColor = UIManager.getLookAndFeelDefaults().getColor("Focus.color");
```

Comment:
I guess we normally call like this in Basic L&F which is extended by different L&Fs so that it will pick up the defaults from the particular L&F in question, otherwise UIManager.getColor(). should suffice as Focus.color is defined in AquaLookAndFeel. 
But I am not sure with this hardcoded values..Can't we leverage viewRect or textRect to get the required coordinates?

Reply:
> UIManager.getColor(). should suffice

Fixed.

> But I am not sure with this hardcoded values..Can't we leverage viewRect or textRect to get the required coordinates?

viewRect is set to be exactly (0, 0, b.width, b.height). Insets are buried deep inside JRS classes and used as hardcoded valued, i can not  get them from there and both textRect and iconRect are not representing what needs to be drawn to be consistent with the way OS draws focus. Plus they are being initialized only later down the code and at the time of this call they are empty. Not that it would matter.